sys/include/auto_init_utils.h

```diff
@@ -69,30 +69,46 @@ typedef struct {
 
 #if IS_ACTIVE(CONFIG_AUTO_INIT_ENABLE_DEBUG) || defined(DOXYGEN)
 /**
- * @brief   Add a module to the auto-initialization array
+ * @brief   Add a module to an auto-initialization array @p xfa
  *
+ * @param   xfa         An XFA
  * @param   function    Function to be called on initialization @ref auto_init_fn_t
  * @param   priority    Priority level @ref auto_init_prio_t
  */
-#define AUTO_INIT(function, priority)                                                   \
-    XFA_CONST(auto_init_xfa, priority)                                                  \
-    auto_init_module_t auto_init_xfa_ ## function                                       \
+#define _AUTO_INIT(xfa, function, priority)                                             \
+    XFA_CONST(xfa, priority)                                                            \
+    auto_init_module_t xfa ## _ ## function                                             \
         = { .init = (auto_init_fn_t)function,                                           \
             .prio = priority,                                                           \
             .name = XTSTR(function) }
 #else
-#define AUTO_INIT(function, priority)                                                   \
-    XFA_CONST(auto_init_xfa, priority)                                                  \
-    auto_init_module_t auto_init_xfa_ ## function                                       \
+#define _AUTO_INIT(xfa, function, priority)                                             \
+    XFA_CONST(xfa, priority)                                                            \
+    auto_init_module_t xfa ## _ ## function                                             \
         = { .init = (auto_init_fn_t)function }
 #endif
 
+/**
+ * @brief   Add a module to the auto-initialization array
+ *
+ * @param   function    Function to be called on initialization @ref auto_init_fn_t
+ * @param   priority    Priority level @ref auto_init_prio_t
+ */
+#define AUTO_INIT(function, priority)       _AUTO_INIT(auto_init_xfa, function, priority)
+
 /**
  * @brief   Construct a priority value equal to @p priority + 1,
  *          to be used with @ref AUTO_INIT
  */
 #define AUTO_INIT_PRIORITY_AFTER(priority)  RIOT_PP_SUCCESSOR(priority)
 
+/**
+ * @brief   Call the auto-init function of module @p module
+ *
+ * @param[in]   module      Module to be initialized
+ */
+void auto_init_module(const volatile auto_init_module_t *module);
```

Comment:
Oh I think `DEBUG()` is a bit in the way to this. Because I would have to `#include debug.h` in `auto_init_utils.h` and ... if this include passes through to some C file which include `auto_init_utils.h` before `debug.h` it could cause confusion.
Or if I don't include `debug.h` I would insist on including `debug.h` before `auto_init_utils.h`.
Or .... I simply change `DEBUG()` to `printf()`.